extensions/Dockerfile

```diff
@@ -0,0 +1,25 @@
+## Builds the webserver for the repo service.
+FROM golang as gobuild
```

Comment:
I am not sure if I am missing some go magic but when using that image or a newer one to build I get:
```
Trying to pull registry.ci.openshift.org/ocp/builder:rhel-8-golang-1.18-openshift-4.11...
Getting image source signatures
Copying blob 47aa3ed2034c skipped: already exists  
Copying blob eac1b95df832 skipped: already exists  
Copying blob 1cfc2c7a4c70 done  
Copying blob 455a09169fee done  
Copying config 3122dc0047 done  
Writing manifest to image destination
Storing signatures
[1/4] STEP 2/5: ADD repo-server/main.go .
[1/4] STEP 3/5: RUN mkdir /build
[1/4] STEP 4/5: RUN go build -o /build/webserver main.go
panic: internal error: rs.graph is unexpectedly nil with -mod=vendor

goroutine 1 [running]:
cmd/go/internal/modload.updatePrunedRoots({0xadb6d8, 0xc0000280a0}, 0x40fc1b?, 0xc000020280, {0xc000300400, 0x79, 0x0?}, {0x0, 0x0, 0x0}, ...)
        /usr/lib/golang/src/cmd/go/internal/modload/buildlist.go:955 +0xb6f
cmd/go/internal/modload.updateRoots({0xadb6d8?, 0xc0000280a0?}, 0x40ecc5?, 0x7f3d53146a68?, {0xc000300400?, 0x400?, 0xc0000cd1b8?}, {0x0, 0x0, 0x0}, ...)
        /usr/lib/golang/src/cmd/go/internal/modload/buildlist.go:653 +0xb7
cmd/go/internal/modload.(*loader).updateRequirements(0xc00012b5f0, {0xadb6d8, 0xc0000280a0})
        /usr/lib/golang/src/cmd/go/internal/modload/load.go:1361 +0x2a5
cmd/go/internal/modload.loadFromRoots({0xadb6d8, 0xc0000280a0}, {{{0x0, 0x0}, 0xc00012c450, 0x0, {0x0, 0x0}, 0x0, 0x1, ...}, ...})
        /usr/lib/golang/src/cmd/go/internal/modload/load.go:1096 +0x70f
cmd/go/internal/modload.ImportFromFiles({0xadb6d8, 0xc0000280a0}, {0xc0000200e0, 0x1, 0x1})
        /usr/lib/golang/src/cmd/go/internal/modload/load.go:704 +0x234
cmd/go/internal/load.GoFilesPackage({0xadb6d8, 0xc0000280a0}, {0x38?, 0x3?, 0xa4?}, {0xc0000200e0?, 0x1, 0x1})
        /usr/lib/golang/src/cmd/go/internal/load/pkg.go:2925 +0x22f
cmd/go/internal/load.PackagesAndErrors({0xadb6d8?, 0xc0000280a0?}, {0xb?, 0x0?, 0x0?}, {0xc0000200e0, 0x1, 0x1})
        /usr/lib/golang/src/cmd/go/internal/load/pkg.go:2673 +0xbf3
cmd/go/internal/work.runBuild({0xadb6d8, 0xc0000280a0}, 0xc0000243f0?, {0xc0000200e0, 0x1, 0x1})
        /usr/lib/golang/src/cmd/go/internal/work/build.go:382 +0xa7
main.invoke(0xdb62e0, {0xc0000200b0, 0x4, 0x4})
        /usr/lib/golang/src/cmd/go/main.go:218 +0x2ee
main.main()
        /usr/lib/golang/src/cmd/go/main.go:175 +0x78e
error building at STEP "RUN go build -o /build/webserver main.go": error while running runtime: exit status 2
ERRO[0045] exit status 2   ```

I changed back to a `golang` base image using go 1.17 and 1.18 instead of 1.18 and they still work. I also changed my Dockerfile to use Centos8 stream to build too not just create the repo and that works as well. I updated the PR with that last version.

Reply:
Humm based on: https://maelvls.dev/go111module-everywhere/ and https://github.com/golang/go/issues/16562 I think [this](https://github.com/openshift/machine-config-operator/blob/5516bd8911ce3e53cfc20924609c129d2eabc604/Makefile#L17) is manipulating the default value on the image on the MCO build.  I will give that a try. But... @cgwalters does it make sense to build and create the repo on the same centos container either way?

Reply:
Let's figure this one out before merging

Reply:
I just used stream8 to also build the go binary which gives one one less stage in my latest push.